src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -3177,19 +3177,22 @@ public function syncOriginalAttribute($attribute)
      */
     public function isDirty($attributes = null)
     {
-        $dirty = $this->getDirty();
-
         if (is_null($attributes)) {
-            return count($dirty) > 0;
-        }
-
-        if (! is_array($attributes)) {
+            $attributes = array_keys($this->attributes);
+        } elseif (! is_array($attributes)) {
             $attributes = func_get_args();
         }
 
-        foreach ($attributes as $attribute) {
-            if (array_key_exists($attribute, $dirty)) {
-                return true;
+        foreach ($attributes as $key) {
+            if (isset($this->attributes[$key])) {
```

Comment:
Here you are checking with ```isset``` so the key must exists and the value must not be ```null```. The original check is different to this. Intentional?